pulser/simulation/simconfig.py

```diff
@@ -0,0 +1,129 @@
+        """
+    def __init__(self, noise_types: List[str] = [], runs: int = 15,
+                 samples_per_run: int = 5, temperature: float = 50.,
+                 laser_waist: float = 175.,
+                 solver_options: qutip.Options = qutip.Options(max_step=5)):
+        self.noise = noise_types
+        self.temperature = temperature
+        self.runs = runs
+        self.samples_per_run = samples_per_run
+        self.laser_waist = laser_waist
+        self.solver_options = solver_options
+        self.init_spam()
+
+    def __str__(self) -> str:
+        s = ""
+        s += "Options:\n"
+        s += "----------       \n"
+        s += "noise types:       " + str(self.noise) + "\n"
+        s += "spam dictionary:   " + str(self.spam_dict) + "\n"
+        s += "temperature:       " + str(self.temperature) + "K" + "\n"
+        s += "number of runs:    " + str(self.runs) + "\n"
+        s += "samples per runs:  " + str(self.samples_per_run) + "\n"
+        s += "laser waist:       " + str(self.laser_waist) + "μm \n"
+        s += "\n" + "Solver Options: \n"+str(self.solver_options)[10:-1]+"\n"
+        return s
+
+    @property
+    def noise(self) -> List[str]:
+        return self._noise
+
+    @noise.setter
+    def noise(self, noise_types: List[str]) -> None:
+        self._noise: List[str] = []
+        for noise_type in noise_types:
+            self._add_noise(noise_type)
+
+    def _add_noise(self, noise_type: str) -> None:
+        """Adds a noise model to the SimConfig instance, to be used
+            in Simulation.
+            Args:
+                noise_type (str): Choose among:
+                    'dephasing': random phase (Z) flip
+                    'doppler': Noisy doppler runs
+                    'amplitude': Noisy gaussian beam
+                    'SPAM': SPAM errors. Adds:
+                        eta: Probability of each atom to be badly prepared
+                        epsilon: false positives
+                        epsilon_prime: false negatives
+        """
+        # Check proper input:
+        if noise_type not in noise_dict_set:
+            raise ValueError('Not a valid noise type')
+        self._noise.append(noise_type)
+
+    @property
+    def spam_dict(self) -> dict[str, float]:
+        return self._spam_dict
+
+    def set_spam(self, **values: float) -> None:
+        """Allows the user to change SPAM parameters in dictionary"""
+        for param in values:
+            if param not in {'eta', 'epsilon', 'epsilon_prime'}:
+                raise ValueError('Not a valid SPAM parameter')
+            if values[param] > 1 or values[param] < 0:
+                raise ValueError('Invalid value : must be between 0 and 1')
+            self._spam_dict[param] = values[param]
+
+    def init_spam(self) -> None:
+        self._spam_dict: dict[str, float] = \
+            {'eta': 0.005, 'epsilon': 0.01, 'epsilon_prime': 0.05}
+
+    @property
+    def temperature(self) -> float:
```

Comment:
I was not sure if `dataclass` would work with a `setter`, but I tested it and it works

Reply:
It doesn't work for me : when I try to set the `temperature` field in a dataclass, I get : 
`    101     def temperature(self, value: float) -> None:
    102         # value set in microkelvin
--> 103         self.temperature = value * 1e-6
    104         self._calc_sigma_doppler()
    105 

TypeError: unsupported operand type(s) for *: 'property' and 'float' `

Reply:
I think that's because you have `self.temperature` instead of `self._temperature`

Reply:
None seem to work. I'm not all keen on using `properties` with `dataclasses`, it seems it's quite a hassle from what I've been researching. Maybe we should stick to a regular class.

Reply:
I find that odd, a `dataclass` is just a regular class with a default `__init__` function, using a `property` shouldn't be any different. Can you point me to your research on this?

Reply:
https://florimond.dev/en/posts/2018/10/reconciling-dataclasses-and-properties-in-python/
I could try doing this but it kinda feels like a weird workaround.